naomi-setup.sh

```diff
@@ -7,54 +7,239 @@
+    cd .. # ~/.naomi/local
 
-# install setuptools and pip for package management
-echo 'Installing setuptools'
-wget https://files.pythonhosted.org/packages/37/1b/b25507861991beeade31473868463dad0e58b1978c209de27384ae541b0b/setuptools-40.6.3.zip
-unzip setuptools-40.6.3.zip
-cd setuptools-40.6.3
-~/.naomi/local/bin/python setup.py install  # specify the path to the python you installed above
-cd .. # ~/.naomi/local
+    # The old version of pip expects to access pypi.org using http. PyPi now
+    # requires ssl for all connections.
+    NAME=pip
+    VERSION=18.1
+    VERNAME=$NAME-$VERSION
+    CHKSUM=c0a292bd977ef590379a3f05d7b7f65135487b67470f6281289a94e015650ea1
+    TARFILE=$VERNAME.tar.gz
+    URL=https://files.pythonhosted.org/packages/45/ae/8a0ad77defb7cc903f09e551d88b443304a9bd6e6f124e75c0fbbf6de8f7/$TARFILE
+    if [ ! -f $TARFILE ]; then
+        wget $URL
+    fi
+    echo "$CHKSUM $TARFILE" | sha256sum -c -
+    if [ $? -eq 0 ]; then
+        echo "$TARFILE checksum verified"
+    else
+        echo "$TARFILE checksum failed" >&2
+        exit 1
+    fi
+    tar xvzf $TARFILE
+    cd $VERNAME # ~/.naomi/local/pip-18.1
+    ~/.naomi/local/bin/python setup.py install  # specify the path to the python you installed above
 
-# The old version of pip expects to access pypi.org using http. PyPi now
-# requires ssl for all connections.
-wget https://files.pythonhosted.org/packages/45/ae/8a0ad77defb7cc903f09e551d88b443304a9bd6e6f124e75c0fbbf6de8f7/pip-18.1.tar.gz
-tar xvzf pip-18.1.tar.gz
-cd pip-18.1 # ~/.naomi/local/pip-18.1
-~/.naomi/local/bin/python setup.py install  # specify the path to the python you installed above
+    # install naomi & dependancies
+    echo "Returning to $NAOMI_DIR"
+    cd $NAOMI_DIR
+    ~/.naomi/local/bin/pip install -r python_requirements.txt
 
-# install naomi & dependancies
-echo "Returning to $NAOMI_DIR"
-cd $NAOMI_DIR
-~/.naomi/local/bin/pip install -r python_requirements.txt
+    # start the naomi setup process
+    echo "#!/bin/bash" > Naomi
+    echo "~/.naomi/local/bin/python $NAOMI_DIR/Naomi.py \$@" >> Naomi
+fi
+if [ $OPTION = "3" ]; then
+    if [ $APT -eq 1 ]; then
+        echo 'Making sure you have the latest python3 and pip3 installed on your system'
+        sudo apt-get install python3 python3-pip
+    else
+        echo "Please ensure that python3 and pip3 are installed on your system"
+    fi
+    pip3 install -r python_requirements.txt
+    # start the naomi setup process
+    echo "#!/bin/bash" > Naomi
+    echo "python3 $NAOMI_DIR/Naomi.py \$@" >> Naomi
+fi
 ./compile_translations.sh
-
-# start the naomi setup process
-echo "#!/bin/bash" > Naomi
-echo "~/.naomi/local/bin/python $NAOMI_DIR/Naomi.py \$@" >> Naomi
 chmod a+x Naomi
+if [ $OPTION = "1" ]; then
+    echo "You will need to activate the Naomi virtual environment when installing"
+    echo "or testing python modules for Naomi using the following command:"
+    echo "  $ workon Naomi"
```

Comment:
For this block of text, add couple of new lines before each block.

Starting with "You will need to activate the Naomi virtual environment when installing"

and after "In the future, run $NAOMI_DIR/Naomi to start Naomi"

![image](https://user-images.githubusercontent.com/3073252/50801038-4b5d8c00-12a0-11e9-905b-6e5afeef87ed.png)

Reply:
Added